spec/factories/edition_factory.rb

```diff
@@ -87,5 +87,15 @@
         )
       end
     end
+
+    trait :withdrawn do
+      summary { SecureRandom.alphanumeric(10) }
+      live { true }
+
+      after(:build) do |edition, evaluator|
+        edition.revision = evaluator.association(:revision)
+        edition.status = create(:withdrawal).status
```

Comment:
We should avoid using a create here as that will always do a db query.

I'd suggest the most consistent way to do what you're aiming for here though would be to have a withdrawn trait on the status factory, so something like:

```
trait :withdrawn do
  state { "withdrawn" }
  association :details, factory: :withdrawal
end
```

And then you can probably create your edition in your test with `create(:edition, live: true, status: build(:status, :withdrawal))`

Or if we have it in edition factory still, the trait could do 

```
      after(:build) do |edition, evaluator|
        edition.status = evaluator.association(
          :status,
          :withdrawal,
          created_by: edition.created_by,
          revision_at_creation: edition.revision,
        )
      end
```

we shouldn't need the revision part in this factory by the way.

Also looking at the polymorphic section on https://github.com/thoughtbot/factory_bot/blob/master/GETTING_STARTED.md#associations made me think there's probably some other ways we can clean up this edition factory.